core/unit_test/TestMathematicalFunctions.hpp

```diff
@@ -53,6 +61,14 @@ template <class T>
 using math_unary_function_return_type_t = typename math_unary_function_return_type<T>::type;
 template <class, class>
 struct math_binary_function_return_type;
+#if defined(KOKKOS_HALF_T_IS_FLOAT) && !KOKKOS_HALF_T_IS_FLOAT
+template <> struct math_binary_function_return_type<KE::half_t, KE::half_t> { using type = KE::half_t; };
+template <> struct math_binary_function_return_type<int, KE::half_t> { using type = KE::half_t; };
```

Comment:
This look fishy

Reply:
You would need to provide much more than that.
Where is it being used?

Reply:
The math binary function test calls. Please add what's missing.

Reply:
do we mix int and half_t with the first argument as int and not the second?